### PR TITLE
Spell: Set missing hook wards in spell

### PR DESCRIPTION
**Pull request by onnovisser**
### Product requirements

ContractUpdate was missing as ward on hooks. Set them in the V2CleanupSpell and in the deployer.

### Design notes

* Needed more changes in the spell than I'd like, due to added chains having a different root address and/or not having v2 root there.

### TODOs

- [ ]

### CODE REVIEW

File: src/spell/V2CleaningsSpell.sol [hunk 1]
```diff
@@ -7,7 +7,12 @@ import {IERC20} from "../misc/interfaces/IERC20.sol";
 import {Root} from "../admin/Root.sol";
 
 Root constant ROOT_V2 = Root(0x0C1fDfd6a1331a875EA013F3897fc8a76ada5DfC);
-Root constant ROOT_V3 = Root(0x7Ed48C31f2fdC40d37407cBaBf0870B2b688368f);
+
+address constant CONTRACT_UPDATER = 0x3B150B19245D2C366bc8f18c775b725DFB298F71;
+address constant FREEZE_ONLY_HOOK = 0xd5B243F05b2906F1f6C80c6096945faADa0731C1;
+address constant FULL_RESTRICTIONS_HOOK = 0x8E680873b4C77e6088b4Ba0aBD59d100c3D224a4;
+address constant FREELY_TRANSFERABLE_HOOK = 0x2a9B9C14851Baf7AD19f26607C9171CA1E7a1A61;
+address constant REDEMPTION_RESTRICTIONS_HOOK = 0xE5423eD8602Fa0F263e17b6212d88Efe42317f06;
```
Comment (lemunozm) [inline src/spell/V2CleaningsSpell.sol:15]: verified ✅

File: test/integration/spell/V2Cleanings.t.sol [hunk 2]
```diff
@@ -105,14 +143,42 @@ contract V2CleaningsSpellTest is Test {
...
+        _testCase(
+            "hyper-evm", string.concat("https://hyperliquid-mainnet.g.alchemy.com/v2/", vm.envString("ALCHEMY_API_KEY"))
+        );
+    }
+
+    function testV2CleaningsMonadMainnet() external {
+        _testCase("monad", string.concat("https://monad-mainnet.g.alchemy.com/v2/", vm.envString("ALCHEMY_API_KEY")));
+    }
+
+    function testV2CleaningsPlumeMainnet() external {
```
Comment (lemunozm) [inline test/integration/spell/V2Cleanings.t.sol:180]: We also need to add it in pharos. It has its own `PHAROS_API_KEY`. I think we can normalize this with `envconfig.rpcUrl()` in the tests, which already has the key on it.

Locally you'll need to fetch the keys using `script/deploy/load_secrets.py` first (only once for project)

**Review by lemunozm — COMMENTED** [review]
The change is surgically clean 👌🏻. All edge cases I could see are solved.

**Comment by lemunozm** [comment]
ward CI job fails because we need to modify `Deployer.t.sol` to reflect these changes there.

File: src/spell/V2CleaningsSpell.sol [hunk 3]
```diff
@@ -113,4 +127,11 @@ contract V2CleaningsSpell {
             ROOT_V2.denyContract(address(ESCROW_V2), address(this));
         }
     }
+
+    function _relyContractUpdaterOnHooks(Root rootV3) internal {
```
Comment (wischli) [inline src/spell/V2CleaningsSpell.sol:133]: Nit, we could check if `code.length > 0` to be fully safe but I agree it's redundant here.

File: test/integration/spell/V2Cleanings.t.sol [hunk 4]
```diff
@@ -4,13 +4,21 @@ pragma solidity 0.8.28;
...
 
+import {Root} from "../../../src/admin/Root.sol";
+
+import {EnvConfig, Env} from "../../../script/utils/EnvConfig.s.sol";
+
 import "forge-std/Test.sol";
 
 import {V2CleaningsSpell} from "../../../src/spell/V2CleaningsSpell.sol";
 import {
     V2CleaningsSpell,
```
Comment (wischli) [inline test/integration/spell/V2Cleanings.t.sol]: Nit: Duplicate important. I'll have Claude investigate why our script is missing this edge-case.

File: test/integration/spell/V2Cleanings.t.sol [hunk 5]
```diff
@@ -33,41 +41,60 @@ contract V2CleaningsSpellTest is Test {
...
         // ----- PRE SPELL -----
 
-        uint256 preTreasuryValue = _usdc().balanceOf(TREASURY);
-        uint256 preEscrowValue = _usdc().balanceOf(ESCROW_V2);
+        IERC20 usdc = _usdc();
+        uint256 preTreasuryValue;
+        uint256 preEscrowValue;
+        if (address(usdc) != address(0)) {
+            preTreasuryValue = _usdc().balanceOf(TREASURY);
+            preEscrowValue = _usdc().balanceOf(ESCROW_V2);
```
Comment (wischli) [inline test/integration/spell/V2Cleanings.t.sol]: Nit: Either use `usdc` var or drop the declaration
```suggestion
            preTreasuryValue = usdc.balanceOf(TREASURY);
            preEscrowValue = usdc.balanceOf(ESCROW_V2);
```

File: src/spell/V2CleaningsSpell.sol (hunk 1 above)
Comment (wischli) [reply src/spell/V2CleaningsSpell.sol:15]: verified

**Review by wischli — COMMENTED** [review]
Don't we need to do this for all the perPool-hooks as well?

**Comment by onnovisser** [comment]
> Don't we need to do this for all the perPool-hooks as well?

@wischli Added it to the pool hooks deploy script as well, though I think we can skip it for existing hooks, unless they need the hookManager functionality
/cc @mustermeiszer

File: test/integration/spell/V2Cleanings.t.sol (hunk 4 above)
Comment (wischli) [reply test/integration/spell/V2Cleanings.t.sol]: Import script fixed in https://github.com/centrifuge/protocol/pull/798

**Review by wischli — DISMISSED** [review]
LGTM!

**Review by wischli — APPROVED** [review]
Re-approving!
